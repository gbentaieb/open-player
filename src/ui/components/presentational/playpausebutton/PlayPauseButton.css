.PlayPauseButton {
  position: relative;
  top: 50%;
  margin-left: 10px;
  margin-right: 10px;
  transform: translateY(-50%);
}
