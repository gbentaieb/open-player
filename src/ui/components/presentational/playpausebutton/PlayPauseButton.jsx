import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Button from '../button/Button';

import { PLAY_GLYPH, PAUSE_GLYPH } from '../../../constants/Glyphs';

import style from './PlayPauseButton.css';

class PlayPauseButton extends Component {
  static propTypes = {
    onClick: PropTypes.func.isRequired,
    isPlaying: PropTypes.bool.isRequired,
  }

  static defaultProps = {
    /* Add default props */
  }

  render() {
    return (
      <div className={style.PlayPauseButton}>
        <Button
          onClick={this.props.onClick}
          glyph={this.props.isPlaying ? PAUSE_GLYPH : PLAY_GLYPH}
        />
      </div>
    );
  }
}

export default PlayPauseButton;
