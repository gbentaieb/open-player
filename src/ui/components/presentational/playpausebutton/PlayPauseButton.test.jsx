import React from 'react';
import PlayPauseButton from './PlayPauseButton';

const defaultProps = {
  onClick: () => {},
  isPlaying: true,
  hoverColor: '00BCD4',
};

describe('PlayPauseButton snapshot', () => {
  test('PlayPauseButton > Snapshot', () => {
    const wrapper = shallow(<PlayPauseButton {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
