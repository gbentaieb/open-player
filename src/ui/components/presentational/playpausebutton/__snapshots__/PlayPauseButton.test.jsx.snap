// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayPauseButton snapshot PlayPauseButton > Snapshot 1`] = `
<div>
  <WithStyles(Button)
    glyph={Object {}}
    onClick={[Function]}
  />
</div>
`;
